ui/test/nuxeo-lock-toggle-button.test.js

```diff
@@ -14,82 +14,318 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-        html`
-          <nuxeo-lock-toggle-button .document=${doc}></nuxeo-lock-toggle-button>
-        `,
-      );
+      await flush();
+      expect(dom(button.root).querySelector('.action')).to.be.null;
+    });
+
+    test('Should not be visible when document is immutable', async () => {
+      const doc = Object.assign({}, lockedDocument);
+      doc.facets = ['Immutable'];
+      button.document = doc;
+      await flush();
+      expect(dom(button.root).querySelector('.action')).to.be.null;
+    });
+
+    test("Should not be visible when document type is 'Root'", async () => {
+      const doc = Object.assign({}, lockedDocument);
+      doc.type = 'Root';
+      button.document = doc;
+      await flush();
+      expect(dom(button.root).querySelector('.action')).to.be.null;
+    });
+
+    test('Should be visible otherwise', async () => {
+      button.document = lockedDocument;
+      await flush();
+
+      const actionDiv = dom(button.root).querySelector('.action');
+      expect(actionDiv).to.not.be.null;
+      expect(actionDiv.querySelector('.label')).to.not.be.null;
+      expect(actionDiv.querySelector('paper-icon-button')).to.not.be.null;
+    });
+
+    test('Should have corresponding information when visible and locked', async () => {
+      button.document = lockedDocument;
+      await flush();
+
+      const actionDiv = dom(button.root).querySelector('.action');
+      expect(actionDiv).to.not.be.null;
+
+      const label = actionDiv.querySelector('.label').textContent.trim();
+      expect(label).to.equal('lockToggleButton.tooltip.unlock');
+
+      const icon = actionDiv.querySelector('paper-icon-button');
+      expect(icon).to.not.be.null;
+      expect(icon.icon).to.equal('nuxeo:lock');
+    });
+
+    test('Should have corresponding information when visible and unlocked', async () => {
+      const doc = Object.assign({}, lockedDocument);
+      doc.lockCreated = null;
+      doc.lockOwner = null;
+      button.document = doc;
+      await flush();
+
+      const actionDiv = dom(button.root).querySelector('.action');
+      expect(actionDiv).to.not.be.null;
+
+      const label = actionDiv.querySelector('.label').textContent.trim();
+      expect(label).to.equal('lockToggleButton.tooltip.lock');
+
+      const icon = actionDiv.querySelector('paper-icon-button');
+      expect(icon).to.not.be.null;
+      expect(icon.icon).to.equal('nuxeo:unlock');
+    });
+  });
+
+  suite('Server Responds with Exceptions', () => {
+    setup(async () => {
+      sinon.spy(button.$.opUnlock, 'execute');
+      sinon.spy(button.$.opLock, 'execute');
+    });
+
+    test('Should trigger notify event when trying to unlock document and permissions error is received', async () => {
       server.respondWith('POST', '/api/v1/automation/Document.Unlock', [
-        200,
+        403,
         { 'Content-Type': 'application/json' },
-        '{"entity-type": "document","uid": "1"}',
+        JSON.stringify({
```

Comment:
This 3rd arg can be a function which we could use to count calls which would allow us to check whether or not the requests were executed without having to use spies.